File: examples/python/batch_calls.py
"""
Example: Create batch calls campaign

PowerShell:
  $env:NEXA_API_KEY='your_key'
  $env:NEXA_ORGANIZATION_ID='encrypted_org_id_base64_string'
  python examples/python/batch_calls.py path/to/leads.csv "My Campaign" agent_123 1

Requires local package path in sys.path:
  cd python && pip install -e .
"""

import os
import sys
import json
from dotenv import load_dotenv

load_dotenv()

# Allow running the example without installing the package
THIS_DIR = os.path.dirname(__file__)
REPO_PYTHON_DIR = os.path.abspath(os.path.join(THIS_DIR, "..", "..", "python"))
if REPO_PYTHON_DIR not in sys.path:
    sys.path.insert(0, REPO_PYTHON_DIR)

from sdk import BatchCallsClient


def main() -> int:
    if len(sys.argv) < 5:
        print(
            "Usage: python examples/python/batch_calls.py <file_path> <title> <agent_id> <agent_version_number>")
        return 1

    file_path = sys.argv[1]
    title = sys.argv[2]
    agent_id = sys.argv[3]
    try:
        agent_version_number = int(sys.argv[4])
    except Exception:
        print("agent_version_number must be an integer")
        return 1

    api_key = os.getenv("NEXA_API_KEY")
    base_url = os.getenv("NEXA_BASE_URL")
    x_org_id = os.getenv("NEXA_ORGANIZATION_ID")

    if not api_key:
        print("NEXA_API_KEY is not set")
        return 1
    if not x_org_id:
        print("NEXA_ORGANIZATION_ID is not set")
        return 1

    client = BatchCallsClient(api_key=api_key, base_url=base_url)
    try:
        resp = client.create_batch_calls(
            file_path=file_path,
            title=title,
            agent_id=agent_id,
            agent_version_number=agent_version_number,
            x_organization_id=x_org_id,
        )
        print(json.dumps(resp.raw or {
            "statusCode": resp.statusCode,
            "message": resp.message,
            "data": {
                "campaign_id": resp.data.campaign_id,
                "total_leads_imported": resp.data.total_leads_imported,
                "campaign_status": resp.data.campaign_status,
            },
        }, indent=2))
        return 0
    except Exception as e:
        print("Error:", e)
        return 1


if __name__ == "__main__":
    raise SystemExit(main())

File: examples/js/batch-calls.mjs
// Example usage for Batch Calls (JS)
// PowerShell:
//   $env:NEXA_API_KEY='your_key'
//   $env:NEXA_ORGANIZATION_ID='encrypted_org_id_base64_string'
//   node examples/js/batch-calls.mjs path/to/leads.csv "My Campaign" agent_123 1
import 'dotenv/config';
import { createBatchCalls } from "../../js/dist/batchCalls.js";
import fs from 'fs';

const apiKey = process.env.NEXA_API_KEY || "";
const baseUrl = process.env.NEXA_BASE_URL || undefined;
const xOrg = process.env.NEXA_ORGANIZATION_ID || "";

const filePath = process.argv[2];
const title = process.argv[3];
const agentId = process.argv[4];
const agentVersionNumber = Number(process.argv[5]);

if (!apiKey) {
  console.error("NEXA_API_KEY is not set");
  process.exit(1);
}
if (!xOrg) {
  console.error("NEXA_ORGANIZATION_ID is not set");
  process.exit(1);
}
if (!filePath || !title || !agentId || !Number.isFinite(agentVersionNumber)) {
  console.error("Usage: node examples/js/batch-calls.mjs <file_path> <title> <agent_id> <agent_version_number>");
  process.exit(1);
}

// Use a ReadStream for Node to ensure proper multipart handling
const stream = fs.createReadStream(filePath);

try {
  const resp = await createBatchCalls({
    apiKey,
    xOrganizationId: xOrg,
    file: stream,
    filename: filePath.split(/[/\\]/).pop(),
    title,
    agentId,
    agentVersionNumber,
    baseUrl,
  });
  console.log(JSON.stringify(resp, null, 2));
} catch (err) {
  console.error("Error:", err?.message || err);
  process.exit(1);
}
